Consumers/StripeMessageConsumer.cs

```diff
@@ -0,0 +1,270 @@
+using Amazon.SQS;
+using Amazon.SQS.Model;
+using Chefster.Common;
+using Chefster.Models;
+using Chefster.Services;
+using Hangfire;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Stripe;
+
+namespace Chefster.Consumers;
+
+public class StripeMessageConsumer(
+    LoggingService loggingService,
+    IServiceScopeFactory serviceScopeFactory,
+    IConfiguration configuration
+) : BackgroundService
+{
+    private readonly LoggingService _logger = loggingService;
+    private readonly AmazonSQSClient _amazonSQSClient = new();
+    private readonly IConfiguration _configuration = configuration;
+    private readonly IServiceScopeFactory _serviceScopeFactory = serviceScopeFactory;
+
+    public async Task MessageConsumer()
+    {
+        var queueUrl = _configuration["CALLBACK_QUEUE"]!;
+        using var scope = _serviceScopeFactory.CreateScope();
+        var _addressService = scope.ServiceProvider.GetRequiredService<AddressService>();
+        var _familyService = scope.ServiceProvider.GetRequiredService<FamilyService>();
+        var _hubSpotService = scope.ServiceProvider.GetRequiredService<HubSpotService>();
+        var _jobService = scope.ServiceProvider.GetRequiredService<JobService>();
+        var _subscriptionService =
+            scope.ServiceProvider.GetRequiredService<Services.SubscriptionService>();
+        var _letterQueueService = scope.ServiceProvider.GetRequiredService<LetterQueueService>();
+
+        while (true)
+        {
+            // Wait up to 20 seconds per interval for messages
+            var receivedMessageRequest = new ReceiveMessageRequest
+            {
+                QueueUrl = queueUrl,
+                MaxNumberOfMessages = 10,
+                WaitTimeSeconds = 20
+            };
+
+            // Handle each message
+            var response = await _amazonSQSClient.ReceiveMessageAsync(receivedMessageRequest);
+            foreach (var message in response.Messages)
+            {
+                try
+                {
+                    var stripeEvent = EventUtility.ParseEvent(message.Body);
+                    Console.WriteLine("[Handling]: " + stripeEvent.Type);
```

Comment:
Probably worth actually logging this along with any other event information we can

Reply:
marking as done